control-plane-operator/controllers/hostedcontrolplane/cno/clusternetworkoperator.go

```diff
@@ -403,26 +467,44 @@ func ReconcileDeployment(dep *appsv1.Deployment, params Params, platformType hyp
 				{Name: "hosted-etc-kube", MountPath: "/etc/hosted-kubernetes"},
 			},
 		},
-
-		// Add an InitContainer that transmutes the in-cluster config to a Kubeconfig
-		// So CNO thinks the "default" config is the hosted cluster
 		{
-			Command: []string{"/bin/bash"},
+			// Token minter InitContainer
+			// This will create the initial token for the CNO to use
+			// then the sidecar will refresh it
+			Command: []string{
+				"/usr/bin/control-plane-operator",
+				"token-minter",
+			},
 			Args: []string{
-				"-c",
-				`
-set -xeuo pipefail
-kc=/configs/management
-kubectl --kubeconfig $kc config set clusters.default.server "https://[${KUBERNETES_SERVICE_HOST}]:${KUBERNETES_SERVICE_PORT}"
-kubectl --kubeconfig $kc config set clusters.default.certificate-authority /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
-kubectl --kubeconfig $kc config set users.admin.tokenFile /var/run/secrets/kubernetes.io/serviceaccount/token
-kubectl --kubeconfig $kc config set contexts.default.cluster default
-kubectl --kubeconfig $kc config set contexts.default.user admin
-kubectl --kubeconfig $kc config set contexts.default.namespace $(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace)
-kubectl --kubeconfig $kc config use-context default`,
+				"--service-account-namespace",
+				workerNamespace,
+				"--service-account-name",
+				workerServiceAccount,
+				"--token-file",
+				fmt.Sprintf("/var/client-token/%s", tokenFile),
+				"--token-audience",
+				params.IssuerURL,
+				"--kubeconfig",
+				fmt.Sprintf("/etc/kubernetes/%s", kas.KubeconfigKey),
+				"--oneshot",
 			},
-			Name:  "rewrite-config",
-			Image: params.Images.CLI,
+			Name:                     fmt.Sprintf("init-%s", tokenMinterContainerName),
+			Image:                    params.Images.TokenMinter,
+			ImagePullPolicy:          corev1.PullIfNotPresent,
+			Resources:                cnoResources,
```

Comment:
It should be fine to simply set the resources here since this doesn't need to be preserved on reconcile (it's an init container)

Reply:
Agree, I've created a default function to get the Resources.Request in support.utils because it's highly used and we can reuse it in many other places.